server/src/test/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsResponseTests.java

```diff
@@ -51,12 +51,11 @@ public void testSerialization() throws Exception {
             String nodeId = randomAlphaOfLength(10);
             ShardRouting shardRouting = TestShardRouting.newShardRouting(shardId, nodeId, randomBoolean(), ShardRoutingState.STARTED);
             clusterSearchShardsGroups[i] = new ClusterSearchShardsGroup(shardId, new ShardRouting[] { shardRouting });
-            DiscoveryNode node = DiscoveryNodeUtils.create(
-                shardRouting.currentNodeId(),
-                new TransportAddress(TransportAddress.META_ADDRESS, randomInt(0xFFFF)),
-                VersionUtils.randomVersion(random())
-            );
-            nodes.add(node);
+            Version nodeVersion = VersionUtils.randomVersion(random());
+            DiscoveryNodeUtils.Builder node = DiscoveryNodeUtils.builder(shardRouting.currentNodeId())
+                .address(new TransportAddress(TransportAddress.META_ADDRESS, randomInt(0xFFFF)))
+                .version(nodeVersion, IndexVersion.getMinimumCompatibleIndexVersion(nodeVersion.id), IndexVersion.fromId(nodeVersion.id));
```

Comment:
For nodeVersion > 8.10, fromId could not give you back the IndexVersion you are expecting. 
I would use `IndexVersionUtils.randomVersion()` if you want to introduce additional randomness, or `IndexVersionUtils.current()` if you want something that covers all.

Nit: I would consider using IndexVersion.ZERO or IndexVersion.MINIMUM_COMPATIBLE for min.

CC @thecoop as he is the expert in this area.

Reply:
Yeah, you can't go from version -> indexversion. Using `IndexVersion.MINIMUM_COMPATIBLE` and `IndexVersion.current()` is the best bet here